telegram/ext/commandhandler.py

```diff
@@ -135,21 +145,21 @@ def check_update(self, update):
         """
         if (isinstance(update, Update) and
                 (update.message or update.edited_message and self.allow_edited)):
-            message = update.message or update.edited_message
+            message = update.effective_message
 
-            if message.text and message.text.startswith('/') and len(message.text) > 1:
-                first_word = message.text_html.split(None, 1)[0]
-                if len(first_word) > 1 and first_word.startswith('/'):
-                    command = first_word[1:].split('@')
-                    command.append(
-                        message.bot.username)  # in case the command was sent without a username
+            if message.entities and message.entities[0].type == MessageEntity.BOT_COMMAND and \
+                    message.entities[0].offset == 0:
+                command = message.text[1:message.entities[0].length]
+                args = message.text.split()[1:]
```

Comment:
Wouldn't it be more correct to do `message.text[len(command):].split()` ?

Reply:
This is the way we always calculated args, And the result should be the same I guess, except for an extra `len()` call in your proposal.

Reply:
Depending on the internals of .split() and telegrams entity calculations, but yeah, you're probably right ^^